src/services/submissionService.ts

```diff
@@ -1,57 +1,92 @@
+        const submissionRef = dbInstance.collection('submissions')
         const snapshot = await submissionRef.doc(id).get()
 
         if (!snapshot.exists) {
             return null
         }
+        
+        const submission = mapDbEntityTogqlEntity(snapshot.data() as DocumentData)
 
-        const convertedEntity = mapDbEntityTogqlEntity(snapshot.data() as DocumentData)
-
-        return convertedEntity
+        return submission
     } catch (error) {
         throw new Error(`Error retrieving submission: ${error}`)
     }
 }
 
 export async function getSubmissions(filters: dbSchema.SubmissionSearchFilters = {}) {
     try {
-        const db = getFirestore()
-        let subRef: Query<DocumentData> = db.collection('submissions')
+        const {
+            googleMapsUrl,
+            healthcareProfessionalName,
+            isUnderReview,
+            isApproved,
+            isRejected,
+            orderBy = [{
+                fieldToOrder: 'createdDate',
+                orderDirection: 'desc'
+            }],
+            limit = 20,
+            createdDate,
+            updatedDate,
+            spokenLanguages
+        } = filters
+
+        let subRef: Query<DocumentData> = dbInstance.collection('submissions')
+
+        if (googleMapsUrl) {
+            subRef = subRef.where('googleMapsUrl', '==', googleMapsUrl)
+        }
 
-        if (filters.googleMapsUrl) {
-            subRef = subRef.where('googleMapsUrl', '==', filters.googleMapsUrl)
+        if (healthcareProfessionalName) {
+            subRef = subRef.where('healthcareProfessionalName', '==', healthcareProfessionalName)
         }
 
-        if (filters.healthcareProfessionalName) {
-            subRef = subRef.where('healthcareProfessionalName', '==', filters.healthcareProfessionalName)
+        if (typeof isUnderReview !== 'undefined') {
+            subRef = subRef.where('isUnderReview', '==', isUnderReview)
         }
 
-        if (typeof filters.isUnderReview !== 'undefined') {
-            subRef = subRef.where('isUnderReview', '==', filters.isUnderReview)
+        if (typeof isApproved !== 'undefined') {
+            subRef = subRef.where('isApproved', '==', isApproved)
         }
 
-        if (typeof filters.isApproved !== 'undefined') {
-            subRef = subRef.where('isApproved', '==', filters.isApproved)
+        if (typeof isRejected !== 'undefined') {
+            subRef = subRef.where('isRejected', '==', isRejected)
         }
 
-        if (typeof filters.isRejected !== 'undefined') {
-            subRef = subRef.where('isRejected', '==', filters.isRejected)
+        if (orderBy && Array.isArray(orderBy)) {
+            orderBy.forEach(order => {
+                if (order) {
+                    subRef = subRef.orderBy(order.fieldToOrder, order.orderDirection)
+                }
+            })
+        }
+
+        if (limit) {
+            subRef = subRef.limit(limit)
+        }
+
+        if (filters.createdDate) {
```

Comment:
this should be `createdDate` not `filters.createdDate`

Reply:
@ermish I noticed this literally a minute before this comment came through. Will update ASAP.